app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -3,180 +3,72 @@
 import org.junit.Test;
 import static org.junit.Assert.*;
 
-import java.math.BigInteger;
-import java.nio.ByteBuffer;
+import android.provider.ContactsContract;
+
 import java.util.ArrayList;
+import java.util.Optional;
 
 public class ParseTest {
 
     @Test
-    public void doesNotExist() throws Exception {
-        String code = "const uint16_t CAN_ID;";
-        Parse test = new Parse(code.toCharArray());
+    public void decodeConsts() throws Exception {
+        Parse test = Parse.parseTextFile("constants.h");
 
-        assertEquals(false, test.getDecoder("CAN_ID").isPresent());
-        assertEquals(false, test.getDecoder("ducky").isPresent());
-    }
+        byte[] one  = {(byte) 0xff};
+        assertEquals("YEET: -1", test.decodeToString(0x502, one));
 
-    @Test
-    public void simpleCheck() throws Exception {
-        String code = "const uint16_t CAN_ID = 0x16;\n" +
-                "const uint16_t CAN_ID2 = 0x32;\n" +
-                "const uint16_t CAN_ID3 = 0x64;";
-        Parse test = new Parse(code.toCharArray());
-
-        assertEquals("const", ((IntegerDecoder) test.getDecoder("CAN_ID").get()).contents.typeQualifer);
-        assertEquals("const", ((IntegerDecoder) test.getDecoder("CAN_ID2").get()).contents.typeQualifer);
-        assertEquals("const", ((IntegerDecoder) test.getDecoder("CAN_ID3").get()).contents.typeQualifer);
-
-        assertEquals("uint16_t", ((IntegerDecoder) test.getDecoder("CAN_ID").get()).contents.payloadDataType);
-        assertEquals("uint16_t", ((IntegerDecoder) test.getDecoder("CAN_ID2").get()).contents.payloadDataType);
-        assertEquals("uint16_t", ((IntegerDecoder) test.getDecoder("CAN_ID3").get()).contents.payloadDataType);
-
-        assertEquals("0x16", ((IntegerDecoder) test.getDecoder("CAN_ID").get()).contents.value);
-        assertEquals("0x32", ((IntegerDecoder) test.getDecoder("CAN_ID2").get()).contents.value);
-        assertEquals("0x64", ((IntegerDecoder) test.getDecoder("CAN_ID3").get()).contents.value);
-    }
+        byte[] eight = {(byte) 0xff};
+        assertEquals("BRUH: 255", test.decodeToString(0x402, eight));
+        assertEquals("YOUNG: -1", test.decodeToString(0x403, eight));
```

Comment:
This looks like a lot of removed tests. Unless those tests are now invalid or redundant, removing tests is generally a bit of a red flag.

Reply:
Yeah, the previous tests were no longer compatible to this new version. Most of the deleted tests were only checking if parser part of the code was doing its job. Making the decoder as stateless as possible came at cost of not storing every detail of a constant or structure such as type qualifier and data type.